static/js/publisher/release/components/releasesTable/droppableRow.js

```diff
@@ -50,7 +50,17 @@ const ReleasesTableDroppableRow = props => {
   const [{ isOver, canDrop, item }, drop] = useDrop({
     accept: DND_ITEM_REVISIONS,
     drop: item => {
-      item.revisions.forEach(r => promoteRevision(r, channel));
+      item.revisions.forEach(r => {
+        const revision = JSON.parse(JSON.stringify(r));
```

Comment:
It starts to be asking to become a `clone` helper or something. We use it a lot already.

Reply:
https://github.com/canonical-web-and-design/snapcraft.io/issues/2400

Reply:
The other though I have regarding this:

The fact that we have to do this (clone and reset/strip progressive data) in drag'n'drop handling means that there is too much stuff in the `revision` object that we pass. Or the logic in the action (`promoteRevision`) should be doing this (ignore any progressive release data).
Drag'n'drop handling should ideally know as little as possible about structure of data. It should only care that on drop it's doing something with 'a thing' that was dragged.